components/automate-ui/src/app/pages/+compliance/+node-credentials/node-credentials-list/node-credential-list.component.ts

```diff
@@ -0,0 +1,136 @@
+import { loading } from 'app/entities/entities';
+import {
+  getAllStatus
+} from 'app/entities/node-credentials/node-credential.selectors';
+import { NodeCredential, NodeCredentialTypes } from 'app/entities/node-credentials/node-credential.model';
+import {
+  allCredentials
+} from 'app/entities/node-credentials/node-credential.selectors';
+
+import { NodeCredentialOrder, SortParams } from './node-credential-list.reducer';
+import { nodeCredentialListState } from './node-credential-list.selectors';
+import { SortNodeCredentialList } from './node-credential-list.actions';
+@Component({
+  selector: 'app-node-credential-list',
+  templateUrl: './node-credential-list.component.html',
+  styleUrls: ['./node-credential-list.component.scss']
+})
+export class NodeCredentialListComponent implements OnInit, OnDestroy {
+  private isDestroyed = new Subject<boolean>();
+  public loading$: Observable<boolean>;
+  public instanceNodeCredentials$: Observable<NodeCredential[]>;
+  public conflictErrorEvent = new EventEmitter<boolean>();
+  public openUserModal = new EventEmitter<boolean>();
+  public list: NodeCredential[] = [];
+  public orderBy: NodeCredentialOrder;
+  public nodeCredentialToDelete: NodeCredential;
+  public createModalVisible = false;
+  public creatingNodeCredential = false;
+  public deleteModalVisible = false;
+  public nodesListLoading = true;
+  public messageModalVisible = false;
+  public nodesListSort = 'last_contact';
+  public nodesListSortOrder = 'DESC';
+  public nodesListFilters = [{ key: 'type', values: ['ssh', 'winrm', 'sudo'] }];
+  public sortBy: string;
+  public params: SortParams;
+
+  constructor(
+    private store: Store<NgrxStateAtom>,
+    private layoutFacade: LayoutFacadeService
+  ) {
+    this.loading$ = store.select(getAllStatus).pipe(map(loading));
+  }
+
+  ngOnInit(): void {
+    this.layoutFacade.showSidebar(Sidebar.Settings);
+    this.getNodeList();
+
+  }
+
+  openCreateModal(): void {
+    this.openUserModal.emit(true);
+  }
+
+  orderFor(column) {
+    return column === this.sortBy ? this.orderBy : 'none';
+  }
+
+  handleSortToggle({ detail: sortParams }) {
+
+    this.store.dispatch(new SortNodeCredentialList(sortParams));
+    this.getNodeList();
+  }
+
+  formatKeyType(credentialType: NodeCredentialTypes): string {
+    switch (credentialType) {
+      case NodeCredentialTypes.SSH:
+        return 'SSH';
+      case NodeCredentialTypes.WinRM:
+        return 'WinRM';
+      case NodeCredentialTypes.Sudo:
+        return 'Sudo';
+      default:
+        return 'SSH';
+    }
+  }
+
+  startNodeCredentialDelete($event: MatOptionSelectionChange, p: NodeCredential): void {
+    if ($event.isUserInput) {
+      this.nodeCredentialToDelete = p;
+      this.deleteModalVisible = true;
+    }
+  }
+
+  getNodeList() {
+    this.instanceNodeCredentials$ = this.store.pipe(select(allCredentials));
+    this.store.select(nodeCredentialListState).pipe(
```

Comment:
This subscription needs to be only invoked once; you are invoking it multiple times through multiple calls of `getNodeList`. Need to separate out that setup code from other code in this function (like the dispatch).